osc_sdk/sdk.py

```diff
@@ -255,6 +262,18 @@ def check_options(self):
             abort('Endpoint is not configured')
         if self.method not in METHODS_SUPPORTED:
             abort('Method {} is not supported'.format(self.method))
+        if self.ssl_engine_id:
+            if self.protocol != 'https':
+                abort('Protocol must be "https" with ssl_engine_id')
+            if not self.x509_client_cert:
+                abort('Missing x509_client_cert with ssl_engine_id')
+            if not self.x509_client_key:
+                abort('Missing x509_client_key with ssl_engine_id')
+        else:
```

Comment:
This check is kinda odd because you can not have `self.ssl_engine_id` without `self.ssl_engine_path` but you can have `self.ssl_engine_path` without `self.ssl_engine_id`.
maybe check that if one of those is here all 4 arguments are here. (maybe using a tmp flag)

Reply:
agree this is odd, `ssl_engine_path` and `ssl_module_path` should always be optional